### DISCUSSION TITLE
Enabling OAuth App for IntelliJ Idea integration

### DISCUSSION THREAD

**Discussion by ivanzlenko** [apache/ozone#7351] [Ideas]
IntelliJ Idea made great improvements in their integration with GitHub, however we can't use it right now because OAuth App is not enabled for Idea. Can we enable this integration for Ozone project? 
It will really help with code reviews.

**Comment by errose28** [comment]
Can you clarify what steps would need to be performed from the Ozone project side? Looking at the [Intellij Docs](https://www.jetbrains.com/help/idea/github.html#register-account) it looks like this just has to be done with your personal GitHub account.

I agree having code checked out locally helps a lot with reviews. I usually use GitHub CLI's `gh pr checkout` command for this though so I'm not very familiar with the Intellij integration.

**Reply by ivanzlenko** [reply]
@errose28 I need to be a member of Apache to complete all necessary steps. All steps can be found in [GitHub Doc](https://docs.github.com/en/account-and-profile/setting-up-and-managing-your-personal-account-on-github/managing-your-membership-in-organizations/requesting-organization-approval-for-oauth-apps)

Whilst I have "JetBrains IDE Integration" in my account however if I would drop into it I have only organizations which I'm member of. 

> I usually use GitHub CLI's gh pr checkout command for this though so I'm not very familiar with the Intellij integration.

This integration eliminates necessity for this step. You can checkout to the PR fork straight from IDE itself. Plus it is very easy just to navigate PR and switch to desired file right from it. Plus their Split View for diffs slightly better. 
Adding comments could be slightly panful though since it lacks GUI for formatting, though it is manageable if you remember syntax for formatting options.

**Reply by errose28** [reply]
The restrictions look to be on an organization level, so this would need to be handled by ASF Infra. Even PMC members do not have access to the `Settings` tab of the apache/ozone repo. It looks like this has been brought up in [INFRA-25935](https://issues.apache.org/jira/browse/INFRA-25935) and [INFRA-20618](https://issues.apache.org/jira/browse/INFRA-20618). Both of the tickets seem to suggest using a personal access token instead of OAuth. Can you try this approach?

**Reply by errose28** [reply]
So I was curious about this feature and just tried it myself, but I'm seeing that even using a personal access token against my own Ozone fork independent of Apache I get the same error as shown in this screenshot. The token seems legit since it was able to log me in, and I checked that it has all the required permissions Intellij needs. This kind of looks like an Intellij or GitHub issue now.
![Screenshot 2024-11-07 at 3 38 45 PM](https://github.com/user-attachments/assets/1094fe0b-4157-4d5e-bd06-e9acf4126edb)

**Reply by ptlrs** [reply]
I have enabled access from Intellij to Github via a personal access token. 
I can see the PRs as well as have the capability to checkout those PRs. 
<img width="488" alt="Screenshot 2024-11-11 at 10 46 06 AM" src="https://github.com/user-attachments/assets/f6272674-3c6c-43b5-92f6-ce737c1ed4fd">
<img width="514" alt="Screenshot 2024-11-11 at 10 46 39 AM" src="https://github.com/user-attachments/assets/5a8f04ca-f367-4cb8-bc81-fa51cfee7271">

@errose28 You could try creating a new [classic token](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/managing-your-personal-access-tokens#creating-a-fine-grained-personal-access-token) (as opposed to the newer fine-grained token). 
I have the following privileges enabled for my token:
- repo
- workflow
- read:org
- read:user
- read:email

**Reply by errose28** [reply]
Hmm my token is classic as well. Good to see it's possible to get it working though. I might invest in some troubleshooting on my end at a later time.

**Reply by ivanzlenko** [reply]
@errose28 thanks, will give it a go!

**Reply by ivanzlenko** [reply]
Yeah, steps from https://issues.apache.org/jira/browse/INFRA-20618 looks like helped me to get it working.
